server/web/routes/proposals.js

```diff
@@ -322,6 +323,51 @@ async function getObjectFromS3(fileName) {
   });  
 }
 
+async function findAllProposalsInSameGroup(proposal, includeSelf = false) {
+  const history = [];
```

Comment:
We don't need a separate history list as what we really care about is not a history, we just care if a proposal is original or not and if not, whether it has has uploaded revised proposals. remove it.